### PR TITLE
[Accessibility] Updated the simulator to support accessibility feature

### PR DESCRIPTION
**Pull request by veler**
Added the support of the keyboard navigation and screen reader to the Adafruit simulator.

Here is a footage of the simulator when interactive elements have the focus : 
capture

## Important

This PR also need the following PR from PXT-Core to build : [https://github.com/Microsoft/pxt/pull/2888](https://github.com/Microsoft/pxt/pull/2888)

### CODE REVIEW

File: sim/visuals/board.ts [hunk 1]
```diff
@@ -506,6 +514,14 @@ namespace pxsim.visuals {
...
                 switchSlide.removeAttribute("transform");
             }
+
+            this.updateSwitchAria();
+        }
+
+        private updateSwitchAria() {
+            let status = this.board.slideSwitchState.isLeft() ? "On" : "Off";
+            accessibility.setAria(this.slideSwitch, "button", "On/Off Switch. Current state : " + status);
+            this.slideSwitch.setAttribute("aria-pressed", this.board.slideSwitchState.isLeft().toString());
```
Comment (microsoftsam) [inline sim/visuals/board.ts]: Could you rename this from updateSwitchAria to renderSlideSwitchARIA or something of the like. Only reason I suggest that, is we use the updateFUNC paradigm in this file to indicate we're updating a simulator component, and in this case we're not updating the slide switch state, but we're rendering it's aria components.

Comment (veler) [inline sim/visuals/board.ts]: Sure

File: sim/visuals/pincontrol.ts [hunk 2]
```diff
@@ -15,6 +15,10 @@ namespace pxsim.visuals {
             svg.addClass(this.outerElement, "sim-pin-touch");
             this.addButtonEvents();
 
+            if ((this.pin as pins.CommonPin).used) {
+                accessibility.makeFocusable(this.outerElement);
+                accessibility.setAria(this.outerElement, "button", this.outerElement.firstChild.textContent);
```
Comment (microsoftsam) [inline sim/visuals/pincontrol.ts]: Check that firstChild and textContent exist.

Comment (veler) [reply sim/visuals/pincontrol.ts]: ok

**Comment by microsoftsam** [comment]
Border is too thick also. Can't see the pins when focused on.

**Pull request closed without merging**